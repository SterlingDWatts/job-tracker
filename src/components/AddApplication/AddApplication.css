.AddApplication form {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.AddApplication button {
  padding: 8px;
  border: none;
  border-radius: 4px;
  color: white;
  background-color: #21b0bf;
  min-width: 175px;
  font-size: 16px;
}
