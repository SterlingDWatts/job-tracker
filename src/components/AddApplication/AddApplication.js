import React, { useState, useContext } from "react";
import { useHistory } from "react-router-dom";
import { v4 as uuidv4 } from "uuid";
import { Overlay, LabelGroup, Input, Label, Button } from "../Utils/Utils";
import ApplicationsContext from "../../contexts/ApplicationsContext/ApplicationsContext";
import "./AddApplication.css";

export default function AddApplication(props) {
  const history = useHistory();
  const context = useContext(ApplicationsContext);
  const [form, setForm] = useState({
    company: {
      value: "",
      touched: false,
    },
    companySite: {
      value: "",
      touched: false,
    },
    jobTitle: {
      value: "",
      touched: false,
    },
  });

  function handleChange(e) {
    const { name, value } = e.target;
    setForm({
      ...form,
      [name]: { value, touched: true },
    });
  }

  function handleSubmit(e) {
    e.preventDefault();
    const id = uuidv4();
    const app = {
      id,
      column: [{ id: 0, date: new Date() }],
      company: form.company.value,
      job: form.jobTitle.value,
      site: form.companySite.value,
    };
    const { applications, setApplications } = context;
    setApplications([...applications, app]);
    setForm({
      company: {
        value: "",
        touched: false,
      },
      companySite: {
        value: "",
        touched: false,
      },
      jobTitle: {
        value: "",
        touched: false,
      },
    });
    history.push(`/?app=${id}`);
  }

  return (
    <Overlay className="AddApplication">
      <form onSubmit={handleSubmit}>
        <LabelGroup>
          <Label htmlFor="company">Company</Label>
          <Input
            type="text"
            id="company"
            name="company"
            placeholder="ex: Mindera"
            value={form.company.value}
            onChange={(e) => handleChange(e)}
          />
        </LabelGroup>
        <LabelGroup>
          <Label htmlFor="companySite">Company Site</Label>
          <Input
            type="url"
            id="companySite"
            name="companySite"
            value={form.companySite.value}
            onChange={(e) => handleChange(e)}
          />
        </LabelGroup>
        <LabelGroup>
          <Label htmlFor="jobTitle">Job Title</Label>
          <Input
            type="text"
            id="jobTitle"
            name="jobTitle"
            value={form.jobTitle.value}
            onChange={(e) => handleChange(e)}
          />
        </LabelGroup>
        <Button type="submit">Save</Button>
      </form>
    </Overlay>
  );
}
