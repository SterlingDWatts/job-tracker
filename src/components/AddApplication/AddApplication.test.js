import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { BrowserRouter } from "react-router-dom";
import AddApplication from "./AddApplication";

describe("AddApplication Component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <AddApplication />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const wrapper = shallow(<AddApplication />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
