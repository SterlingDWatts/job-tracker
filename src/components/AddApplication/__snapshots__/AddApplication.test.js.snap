// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddApplication Component renders the UI as expected 1`] = `
<Overlay
  className="AddApplication"
  ellipsesClick={[Function]}
>
  <form
    onSubmit={[Function]}
  >
    <LabelGroup>
      <Label
        htmlFor="company"
      >
        Company
      </Label>
      <Input
        id="company"
        name="company"
        onChange={[Function]}
        placeholder="ex: Mindera"
        type="text"
        value=""
      />
    </LabelGroup>
    <LabelGroup>
      <Label
        htmlFor="companySite"
      >
        Company Site
      </Label>
      <Input
        id="companySite"
        name="companySite"
        onChange={[Function]}
        type="url"
        value=""
      />
    </LabelGroup>
    <LabelGroup>
      <Label
        htmlFor="jobTitle"
      >
        Job Title
      </Label>
      <Input
        id="jobTitle"
        name="jobTitle"
        onChange={[Function]}
        type="text"
        value=""
      />
    </LabelGroup>
    <button
      type="submit"
    >
      Save
    </button>
  </form>
</Overlay>
`;
